### PR TITLE
make NodeList implement List

### PR DESCRIPTION
**Pull request by DeepSnowNeeL**
Here we go

### CODE REVIEW

File: javaparser-core/src/main/java/com/github/javaparser/ast/NodeList.java [hunk 1]
```diff
@@ -161,4 +171,222 @@ public Node getParentNodeForChildren() {
...
+
+    /**
+     * @param o
+     * @return
+     * @see java.util.List#remove(java.lang.Object)
+     */
+    @Override
+    public boolean remove(Object o) {
+        return innerList.remove(o);
+    }
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/ast/NodeList.java:226]: Hmmm, does this need de-parenting?

File: javaparser-core/src/main/java/com/github/javaparser/ast/NodeList.java [hunk 2]
```diff
@@ -161,4 +171,222 @@ public Node getParentNodeForChildren() {
...
+
+    /**
+     * @param c
+     * @return
+     * @see java.util.List#removeAll(java.util.Collection)
+     */
+    @Override
+    public boolean removeAll(Collection<?> c) {
+        return innerList.removeAll(c);
```
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/ast/NodeList.java]: And here, etc...

Comment (matozoid) [reply javaparser-core/src/main/java/com/github/javaparser/ast/NodeList.java]: Actually, I don't think we do that in other nodes as well, right? Hmmm, that means that the childNodes list will accumulate all kinds of crap when you start modifying the AST...

**Comment by matozoid** [comment]
Ready to merge?

**Comment by matozoid** [comment]
Fixes #550

**Comment by DeepSnowNeeL** [comment]
I think so

**Comment by matozoid** [comment]
Note that I squashed this one too @DeepSnowNeeL

**Pull request merged**